src/trio/_tests/test_threads.py

```diff
@@ -165,7 +167,7 @@ def external_thread_fn() -> None:
         thread.start()
         print("waiting")
         while thread.is_alive():
-            await sleep(0.01)
+            await lowlevel.checkpoint()
```

Comment:
I think it's probably better to use `sleep` here because otherwise we'll be busy looping I think?

Like this isn't a matter of "trio just needs another cycle or two of the available tasks" this is "we need to wait for the thread to start and do its stuff"

This happens 3 other places too.

Reply:
this is also a bug in ruffs implementation, flake8-trio only triggers when the literal is exactly `0`

Reply:
wait no, ruff gave TRIO110, which suggests to await an `Event` instead of looping a sleep. Resolving that by replacing with a checkpoint is ... definitely not the way to go.
Maybe I should widen TRIO110 to also catch checkpoints to make sure people don't confuse them and thinks that because it stops warning on `checkpoint` that it "solves" it

Reply:
I manually changed the 0.02 to a checkpoint because an error was being raised from the sleep there.

Reply:
yeah an Event could plausibly not be possible or make sense in a context, in which case `noqa` would be the solution :)